molgenis-data-validation/src/main/java/org/molgenis/data/validation/meta/EntityTypeValidator.java

```diff
@@ -150,33 +150,41 @@ private static void validateOwnIdAttribute(EntityType entityType, Map<String, At
 			if (ownAttr == null)
 			{
 				throw new MolgenisValidationException(new ConstraintViolation(
-						format("Entity [%s] ID attribute [%s] is not part of the entity attributes", entityType.getId(),
-								ownIdAttr.getName())));
+						format("EntityType [%s] ID attribute [%s] is not part of the entity attributes",
+								entityType.getId(), ownIdAttr.getName())));
 			}
 
 			// Validate that ID attribute data type is allowed
 			if (!AttributeUtils.isIdAttributeTypeAllowed(ownIdAttr))
 			{
 				throw new MolgenisValidationException(new ConstraintViolation(
-						format("Entity [%s] ID attribute [%s] type [%s] is not allowed", entityType.getId(),
+						format("EntityType [%s] ID attribute [%s] type [%s] is not allowed", entityType.getId(),
 								ownIdAttr.getName(), ownIdAttr.getDataType().toString())));
 			}
 
 			// Validate that ID attribute is unique
 			if (!ownIdAttr.isUnique())
 			{
 				throw new MolgenisValidationException(new ConstraintViolation(
-						format("Entity [%s] ID attribute [%s] is not a unique attribute", entityType.getId(),
+						format("EntityType [%s] ID attribute [%s] is not a unique attribute", entityType.getId(),
 								ownIdAttr.getName())));
 			}
 
 			// Validate that ID attribute is not nillable
 			if (ownIdAttr.isNillable())
 			{
 				throw new MolgenisValidationException(new ConstraintViolation(
-						format("Entity [%s] ID attribute [%s] is not a non-nillable attribute", entityType.getId(),
+						format("EntityType [%s] ID attribute [%s] is not a non-nillable attribute", entityType.getId(),
 								ownIdAttr.getName())));
 			}
+
+			// Validate that there is a label if the ID attribute is hidden
+			if (!ownIdAttr.isVisible() && entityType.getLabelAttribute() == null)
```

Comment:
Since the check is in the "if (ownIdAttr != null)" branch of the code, it does not work correctly when abstract entities come into play.

For example:
I cannot upload an abstract entity with a hidden ID and a concrete entity which has a visible label and extends the abstract one.